### DISCUSSION TITLE
why did ctx.params become undefined

### DISCUSSION THREAD

**Discussion by bobwatcherx** [freshframework/fresh#1002] [Q&A] [answered]
i followed this tutorial.  but when I want to take the url parameter it's always undefined
https://fresh.deno.dev/docs/getting-started/fetching-data

```

import { Handlers, PageProps } from "$fresh/server.ts";

interface User {
  login: string;
  name: string;
  avatar_url: string;
}

export const handler: Handlers<User | null> = {
  async GET(_, ctx) {
    const { username } = ctx.params;
    console.log("prams",username)
    const resp = await fetch(`https://api.github.com/users/${username}`);
    if (resp.status === 404) {
      return ctx.render(null);
    }
    const user: User = await resp.json();
    return ctx.render(user);
  },
};
```
i have created a file in routes/api/nice/[id].tsx

and acess the url http://localhost:8000/api/nice/bobwatcherx
params is undefined

**Comment by sylc** [comment, marked as answer]
the file is called `[id].tsx` so the params should have an `id` property, not `username`. 
You are logging `username` which will be undefined.

**Comment by bobwatcherx** [comment]
and how to take parameters and display parameters to html?
```
export const handler: Handlers = {
  async GET(_, ctx) {
    const { edit } = ctx.params;
    return ctx.render(myData)
  },
};
```
now I want to get data as well as get parameter values ​​from the url
```
export default function AboutPage({data}:PageProps,props:PageProps) {
  console.log(props.PageProps) <----- BUT THIS NOT WORK 
}
```

example url http://localhost:3000/sample/MYPARAMS

and i want see MyPARAMS in console terminal
